corehq/form_processor/backends/couch/processor.py

```diff
@@ -77,6 +77,21 @@ def apply_deprecation(cls, existing_xform, new_xform):
         deprecated = XFormDeprecated.wrap(existing_xform.to_json())
         if existing_xform._deferred_blobs:
             deprecated._deferred_blobs = existing_xform._deferred_blobs.copy()
+        else:
+            # in the case we are migrating, we need to populate the deferred blobs
+            # with the existing attachments in couch
+            def _couch_attachments_to_deferred_dict(form, name, attachment_info):
+                content = form.get_db().fetch_attachment(form.orig_id, name)
+                return {
+                    "content": content,
+                    "content_type": attachment_info['content_type'],
+                    "content_length": len(content),
+                }
+            deprecated._deferred_blobs = {
+                name: _couch_attachments_to_deferred_dict(existing_xform, name, attachment_meta)
+                for name, attachment_meta in existing_xform._attachments.items()
```

Comment:
I think we also want this to do the same for existing blobs from the blob db. To do that, this should be `existing_xform.blobs.items()`, but the fetch isn't going to work then. Also, rather than setting `deprecated._deferred_blobs` I think it would make sense to use a non-private API (the `_deferred_blobs` copy above is a bit of a hack). Something like this:
```py
for name, meta in existing_xform.blobs.items():
    deprecated.deferred_put_attachment(
        existing_form.fetch_attachment(name, stream=True),
        name=name,
        content_type=meta.content_type,
        content_length=meta.content_length,
    )
```